core-api/src/main/java/com/optimizely/ab/event/internal/EventBuilder.java

```diff
@@ -61,8 +63,32 @@ public EventBuilder(EventBatch.ClientEngine clientEngine, String clientVersion)
         this.clientEngine = clientEngine;
         this.clientVersion = clientVersion;
         this.serializer = DefaultJsonSerializer.getInstance();
+
+        Runnable r = new Runnable() {
+            public void run() {
+                primeSerializer();
+            }
+        };
+
+        ExecutorService executor = Executors.newCachedThreadPool();
+        executor.submit(r);
+
     }
 
+    private void primeSerializer() {
+        Decision decision = new Decision("", "",
```

Comment:
I'd like to be on the safer side and put this inside a `try/catch`. Worst that can happen then is that the priming fails and the very first API call that sends events will take longer

Reply:
Because it is in its own Runnable thread, worst case is thread dies.